### ISSUE TITLE
Customize rate limiter backoff algorithm

### ISSUE THREAD

**Issue by ryansch** [mperham/sidekiq#2530]
Currently we have to subclass the limiter middleware and replace it in the stack when the server boots.  This should be more easily configurable.

**Comment by mperham** [comment]
Yep, it needs to happen.  Let me think about it; suggestions welcome.

**Comment by ryansch** [comment]
We'll iterate on it and I'll get back to you.

**Comment by mperham** [comment]
First thought: a Proc which returns the seconds to pause.

```ruby
Sidekiq::Limiter.backoff = -> {|worker, job, queue, limiter|
  rand(300)
}
```

**Comment by ryansch** [comment]
Would you be ok exposing `#count` on the limiter?

That `proc` might work out pretty well.  I'll experiment with that interface.  We're trying to gracefully respond to a thundering herd.

**Comment by mperham** [comment]
Currently the count is tracked as a job attribute `job['overrated']`.  The current system does back off randomly over five minutes.  Is that not enough delay for the size of the herd?

**Comment by ryansch** [comment]
I'm referring to the `@count` in `Sidekiq::Limiter::Window` specifically.  The default timeout is actually way too high for our current use case.

**Comment by mperham** [comment]
Oh yeah, totally.  That attribute should have a reader exposed.  Will be in the next release.

**Comment by ryansch** [comment]
`instance_variable_get` thanks you ;-)

**Comment by mperham** [comment]
See mperham/sidekiq-ent#5730435

**Comment by mperham** [comment]
Pushed a first pass.  Will iterate on it.

**Comment by ryansch** [comment]
Thanks this is next on my task list so you might have it solved before I get there.  I'll review what you pushed.

**Comment by mperham** [comment]
This is released, in 0.7.3?

**Comment by ryansch** [comment]
Using it.  Forgot to close the issue.
